### PR TITLE
Multiple owners bug

### PR DESCRIPTION
**Pull request by jimis**
Found while reviewing pull request #2649.

### CODE REVIEW

**Comment by jimis** [comment]
@nickanderson I'll incorporate your test in this one

**Comment by jimis** [comment]
trigger build

File: cf-agent/verify_files_utils.c [hunk 1]
```diff
@@ -3580,93 +3580,97 @@ bool VerifyOwner(EvalContext *ctx, const char *file, const Promise *pp, Attribut
...
     {
-        cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, attr, "Unable to make file belong to an unknown group");
-        *result = PromiseResultUpdate(*result, PROMISE_RESULT_FAIL);
-    }
-
-    if (attr.perms.owners->next == NULL && attr.perms.owners->uid == CF_UNKNOWN_OWNER)  // Only one non.existent item
-    {
-        cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, attr, "Unable to make file belong to an unknown user");
-        *result = PromiseResultUpdate(*result, PROMISE_RESULT_FAIL);
+        /* Change ownership to the first known user in the promised list. */
```
Comment (nickanderson) [inline cf-agent/verify_files_utils.c]: yay for comments

**Comment by jimis** [comment]
trigger build

**Comment by jimis** [comment]
trigger build

**Comment by jimis** [comment]
@nickanderson can you review the last two commits that change your tests?

**Pull request merged**